packages/next/src/client/components/router-reducer/create-initial-router-state.ts

```diff
@@ -105,10 +105,13 @@ export function createInitialRouterState({
       null,
   }
 
-  if (location) {
+  if (process.env.NODE_ENV !== 'development' && location) {
```

Comment:
Does this need to be dev mode rather than NODE_ENV?

Reply:
Currently we gate based on NODE_ENV for anything related to prefetching/development, eg:
https://github.com/vercel/next.js/blob/canary/packages/next/src/client/link.tsx#L665